arrows/ocv/detect_motion_3frame_differencing.h

```diff
@@ -0,0 +1,107 @@
+/*ckwg +29
+ * Copyright 2017 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief OCV motion detection 3-frame difference algorithm impl interface
+ */
+
+#ifndef KWIVER_ARROWS_OCV_THREE_FRAME_DIFFERENCING_H_
+#define KWIVER_ARROWS_OCV_THREE_FRAME_DIFFERENCING_H_
+
+#include <memory>
+
+#include <opencv2/opencv.hpp>
+
+#include <vital/types/timestamp.h>
+#include <vital/vital_config.h>
+#include <vital/algo/detect_motion.h>
+#include <vital/config/config_block.h>
+
+#include <arrows/ocv/kwiver_algo_ocv_export.h>
+
+namespace kwiver {
+namespace arrows {
+namespace ocv {
+
+/// OCV implementation of detect_motion using three-frame differencing
+class KWIVER_ALGO_OCV_EXPORT detect_motion_3frame_differencing
+  : public vital::algorithm_impl<detect_motion_3frame_differencing,
+                                 vital::algo::detect_motion>
+{
+public:
+  PLUGIN_INFO( "detect_motion_3frame_differencing",
+               "OCV implementation of detect_motion using three-frame differencing" )
```

Comment:
The name "detect_motion_3frame_differencing" does not have to be so long. The name only has to be unique over implementations of "detect_motion" algorithms. It could be shortened to "ocv_3frame_diff".

Reply:
+1. That makes sense since it is already and algorithm for motion detection.

Reply:
Changed to ocv_3frame_differencing